test/controllers/ViewArrivalNotificationsControllerSpec.scala

```diff
@@ -31,56 +34,97 @@ import play.api.test.FakeRequest
-
-  private val expectedResult = {
-    ViewArrivalMovement(
-      Map(
-        "01-01-2020" -> {
-          Seq(Movement("test updated", "test mrn", "test name", "test office", "test procedure", "test status", Seq("test actions")))
-        },
-        "02-01-2020" -> {
-          Seq(Movement("test updated", "test mrn", "test name", "test office", "test procedure", "test status", Seq("test actions")))
-        }
+  private val mockReferenceDataConnector = mock[ReferenceDataConnector]
+
+  val localDate: LocalDate = LocalDate.now()
+  val localTime: LocalTime = LocalTime.now()
+  val mockCustomsOfficeResponse: CustomsOffice = CustomsOffice("officeId", "office name", Seq("role1", "role2"), Some("testPhoneNumber"))
+
+  val mockConnectorResponse: Seq[Movement] = {
+    Seq(
+      Movement(
+        localDate,
+        localTime,
+        "test mrn",
+        "test name",
+        "officeId",
+        "normal"
+      ),
+      Movement(
+        localDate,
+        localTime,
+        "test mrn",
+        "test name",
+        "officeId",
+        "normal"
       )
     )
   }
 
+  private val viewMovements: Seq[ViewMovement] = Seq(
+    ViewMovement(localDate,
+      localTime,
+      "test mrn",
+      "test name",
+      "officeId",
+      "office name",
+      "normal"),
+    ViewMovement(localDate,
+      localTime,
+      "test mrn",
+      "test name",
+      "officeId",
+      "office name",
+      "normal")
+  )
+
   "ViewArrivalNotifications Controller" - {
     "return OK and the correct view for a GET" in {
 
       when(mockRenderer.render(any(), any())(any()))
         .thenReturn(Future.successful(Html("")))
 
       when(mockDestinationConnector.getArrivalMovements()(any(), any()))
-        .thenReturn(Future.successful(expectedResult))
+        .thenReturn(Future.successful(mockConnectorResponse))
+
+      when(mockReferenceDataConnector.getCustomsOffice(any())(any(), any())).thenReturn(Future.successful(mockCustomsOfficeResponse))
 
       val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
-        .overrides {
-          bind[DestinationConnector].toInstance(mockDestinationConnector)
-        }
+        .overrides (
+          bind[DestinationConnector].toInstance(mockDestinationConnector),
+          bind[ReferenceDataConnector].toInstance(mockReferenceDataConnector)
+        )
         .build()
 
-      val request = FakeRequest(GET, routes.ViewArrivalNotificationsController.onPageLoad().url)
-      val result = route(application, request).value
+      implicit val feAppConfig: FrontendAppConfig = application.injector.instanceOf[FrontendAppConfig]
 
-      val expectedJson = Json.obj(
-        "declareArrivalNotificationUrl" -> frontendAppConfig.declareArrivalNotificationUrl,
-        "dataRows" -> Json.toJson(expectedResult)
+      val request = FakeRequest(
+        GET,
+        routes.ViewArrivalNotificationsController.onPageLoad().url
       )
```

Comment:
Could we move this up to the top. Minor issue but it could be easier to read if this setup is further up.